pennylane/compiler/qjit_api.py

```diff
@@ -106,7 +105,7 @@ def circuit(theta):
     """
 
     if not available(compiler):
-        raise CompileError(f"The {compiler} package is not installed.")
+        raise CompileError(f"The {compiler} package is not installed.")  # pragma: no cover
```

Comment:
@josh146 @timmysilv with adding "no-cover" only in 2-places for skipping such cases and updating the integration tests, I could get %100 codecov. lmk if this is not a common practice in PL and I can get rid of these `pragma`s.

![Screenshot from 2023-11-10 15-37-17](https://github.com/PennyLaneAI/pennylane/assets/10773383/ee28f0d8-6c8b-4ef7-9b5b-4e86aa8d91a6)

Reply:
I think it's fine for errors like these. otherwise you could mock the `compiler` module with monkeypatch, but it's not the biggest deal